### PR TITLE
Feature/place meal register fields

### PR DESCRIPTION
**Pull request by Roilek**
There are some improvements to do but most of the job of uploading a new meal to the database is done and can be reviewed.

### CODE REVIEW

**Review by khalilacheche — APPROVED** [review]
good implementation overall, only small tweaks needed for the next PR. 
We also have to add the other metadata of the meal (pickup location, status, pickup time)

File: app/src/main/java/com/epfl/neighborfood/neighborfoodandroid/models/Meal.java [hunk 1]
```diff
@@ -33,16 +35,24 @@ public Meal(String name, String shortDescription, String longDescription, int im
...
         this.name = name;
         this.shortDescription = shortDescription;
         this.longDescription = longDescription;
         this.imageId = imageId;
         this.allergens = allergens;
         this.price = price;
-    }
-
-    public Meal(){
+        this.retrievalDate = retrievalDate;
```
Comment (raedbm) [inline app/src/main/java/com/epfl/neighborfood/neighborfoodandroid/models/Meal.java:55]: why didn't you make the retrieval date as an attribute in the database too

File: app/src/main/java/com/epfl/neighborfood/neighborfoodandroid/ui/activities/PlaceMealActivity.java [hunk 2]
```diff
@@ -106,10 +123,31 @@ public void onClick(View v) {
...
+                        Toast.makeText(this,
+                                "No field can be empty!",
+                                Toast.LENGTH_SHORT).show();
+                        fieldsAreNotEmpty = false;
+                        break;
+                    }
+                }
+                if (fieldsAreNotEmpty) {
+                    Intent i = new Intent(getApplicationContext(), MainActivity.class);
+                    MealRepository mealRepository = new MealRepository();
```
[the remark this answers is not in the document]
Comment (raedbm) [reply app/src/main/java/com/epfl/neighborfood/neighborfoodandroid/ui/activities/PlaceMealActivity.java:138]: Yes exactly There is no need for creating the repo here anyways.

**Review by raedbm — APPROVED** [review]
Pretty simple and straightforward. I don't have any comments to add besides to the two small ones that i mentioned

**Pull request merged**